src/picai_prep/dcm2mha.py

```diff
@@ -313,13 +319,8 @@ def resolve_duplicates(self):
                                 serie.write_log(f'Removed by {name} tiebreaker from "{mapping}"')
                                 series.remove(serie)
 
-                # after tiebreakers there are still candidates, select at random
-                if len(series) > 1:
-                    chosen_serie = np.random.choice(series)
-                    for serie in series:
-                        if serie != chosen_serie:
-                            serie.mappings.remove(mapping)
-                            serie.write_log(f'Removed by random selection from "{mapping}"')
+                # after tiebreakers there should not be more than one item left
+                assert len(series) <= 1, f'More than one serie left for mapping "{mapping}"'
```

Comment:
Never assert in production code

`if len(series) <= 1: raise UnexpectedDuplicatesError(mapping)` for example.

Reply:
This assertion only goes wrong when we mess up, so I thought it would be applicable in that case?

Reply:
I argue my thoughts on "no assert in production ever" as follows.

With `assert` you are saying: this might raise during testing, but never during production. _def. assert_: state a fact or belief confidently and forcefully. You don't have the confidence while testing (obviously), but should have it after testing: production.

However, we have this line in production. Therefore, the line should never raise an exception. However, we don't have this confidence, and so it might in special cases raise. Either the line shouldn't exist (we have the confidence) or it should be an exception (we understand the possibility)

Reply:
Exactly why I went for assert hahah! In the current code base, I am confident that assertion can never be false. On that basis, we could remove it altogether. However, future changes are made by humans, and could introduce bugs then, which would hopefully be flagged by this assertion.

Reply:
I'm very confident about future mistakes as well xD

Reply:
I'm on it

Reply:
Here you go :) 
https://github.com/DIAGNijmegen/picai_prep/commit/043d16077f74a01d049aebff5bce104d1bb1425c